bpf/tests/nat_test.h

```diff
@@ -0,0 +1,98 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/* Copyright (C) 2021 Authors of Cilium */
+
+// sample unit test program for functions in nat.h
+// It contains function definitions for testing "snat_v4_track_local".
+// "test_snat_v4_track_local" uses mock functions in conntrack.h. If other
+// functions in nat.h need to be tested, please add the function definitions
+// at the bottom. Do not forget to generate mock libraries for functions in
+// conntrack.h before testing.
+
+#define __BPF_HELPERS_SKB__
+#define __BPF_HELPERS__
+#define ENABLE_IPV4
+#define ENABLE_NODEPORT
+
+#define htons bpf_htons
+#define ntohs bpf_ntohs
+
+#include <stdio.h>
+#include <assert.h>
+
+// Include unity test framework and all the mock libraries.
+#include "unity.h"
+#include "mock/mock_helpers.h"
+#include "mock/mock_helpers_skb.h"
+#include "mock/mock_conntrack_stub.h"
+
+// To avoid conflict between functions in bpf/builtins.h and string.h, define
+// __BPF_BUILTINS__ in conntrack_stub.h to make sure the mock library can be
+// compiled without functions in bpf/builtins.h. Undefine __BPF_BUILTINS__ and
+// include bpf/builtins.h here.
+#undef __BPF_BUILTINS__
+#include "bpf/builtins.h"
+
+#include "bpf/ctx/skb.h"
+#include "node_config.h"
+
+// To use mock library for functions in lib/conntrack.h, lib/conntrack.h must
+// be inlcude as well as mock/mock_conntrack_stub.h.
+#include "lib/conntrack.h"
+
+// Define macros like the followings to make sure the original customized
+// functions are mapped to the mock functions.
```

Comment:
This needs more explanation, for example, if I'm writing a new test, how do I know names of the mocked user function? Will it always be generated in the format of "mock_<func>"?

Reply:
Thanks for pointing it out. Explanation added to both `bpf/tests/nat_test.h` and `bpf/mock/conntrack_stub.h`.